### ISSUE TITLE
Dev Call - Priorities

### ISSUE THREAD

**Issue by leo816** [bisq-network/projects#60]
### Topics

    - Introduction:

       What are Bisq's top Priorities?
     
We all agree on [Misq](https://github.com/bisq-network/misq)/Bisq's 2.0 long-term vision. for that to happen, we need the right developers with enough experience on this area. Bisq contributors should do all they can to attract the right talent.

With this in mind, these are Bisq's top priorities in the short-term:

 **1. Tor. v3:**

- we have @ripcurlx and @sqrrm working on this as the top priority.
- October 15th is the date v2 is disabled so we have to fix it before then

**2. API**

- @ghubstan is in charge of this. The goal is to finish a full working API, with the objective of creating trading bots and improve liquidity. 
- What help do you need to reduce the time of this task? once the tor v3 task is finished It is my assumption that Christoph and Sq will help you with this.

**3. Messaging Problem**

- Some messages aren't going through, the reason is yet unclear.
- Currently no one is assigned to this, Mediator's will continue to send logs to @jmaczx to try and get to the bottom of this.

**4. Mobile app**

- @devinbileck Is the current maintainer of the existing mobile app. The goal is to improve it by enabling it to have more options (move a trade from stage 2 to 3). For that, it seems it will have to be redone from scratch. Devin is unaware of the time he will have available to get this done, which is why he has committed to create a proposal for other developers to take over under his lead.

**5.  UX**

- @flybylow created a prototype for misq https://github.com/bisq-network/projects/issues/57
- @pazza has some great ideas, No one is assigned yet, feel free to put ideas out there.
- Pedro has also amazing concepts. 
_Once again the idea is to make small changes to the existing app that require low effort and that can potentially improve the overall user experience. Enabling new coiners to use bisq as they would use any other high volume exchange._

**6.  XMR**

- Monero is our most traded altcoin, we should learn as much as possible and make sure we are doing what we can to attract as many users. these are the main complaints so far that users have reached out to us for:

1. If the auto-confirm is not activated, some sellers never respond and that can entail to having their funds locked for days. (are they all aware that they can activate this tool on settings?)
2. If the trade were to go to mediation, users have to write down their tx id and tx key to the mediator which is an annoying.
3. Given that trading on Bisq is tantamount to routinely checking if somebody hasn't lowered the price below your level it would be nice to have an option to one-click decresase or increase the offer price by 0.01% instead of manually entering the deviation when editing the offer.

**7.  Payment methods**

- Perhaps won't be a huge driver of volume but if it's a low hanging fruit we should improve.
@pazza I believe you are on this. How many payment methods do you plan on implementing in the next 2 months? what help do you need with this?

**8.  Adding Tether**

- There were mixed feelings about this one, while it probably won't mean an instant rise in volume, It requires low effort and it's the altcoin with the highest trade volume in the market right now.

**- Topics for next call:**

- How can we attract more develpers?
- @devinbileck 's mobile improvement proposal https://github.com/bisq-network/proposals/issues/342 
- @jimmyt : what if we used an already-existing, widely used, well documented, easily scalable decentralized messaging protocol such as Matrix?

<details>

<summary>Click to see the example</summary>

what it would look like:

At 00:00, node A creates a SELL swap offer and publishes their public key, and then disconnects from the internet.
At 01:00, node B connects to the internet, scans the blockchain for new swap offers, and discovers node A's swap offer. Node B takes node A's swap offer, publishes their (node B's) public key, and then posts a message in the Matrix room containing payment method info, encrypted with node A's public key and signed with node B's private key. Then node B disconnects from the internet.
At 02:00, node A connects to the internet, realizes their offer has been taken by node B, and then gets all unread Matrix messages, searching for the payment method info message from node B, decrypting it with their own public key and verifying the signature with node B's public key. Upon success, node A posts a message in the Matrix room containing their payment method info, encrypted with node B's public key and signed with node A's private key. Then node A disconnects from the internet.
At 03:00, node B connects to the internet and gets all unread Matrix messages, searching for the payment method info message from node A, decrypting it with their own public key and verifying the signature with node A's public key. Upon success, the operator of node B can send fiat payment according to the payment method details specified by node A, and then can notify that payment is sent. Then node B disconnects from the internet.
At 04:00, node A connects to the internet, discovers node B has indicated that payment has been sent, and the operator of node A verifies that fiat payment has been received. Then, node A confirms that payment has been received, allowing each swapper to close their half of the swap and receive their stablecoin whenever they wish.
In this way, a swap can be completed from start to finish without the swapping parties ever being online at the same time, and without Bisq/Commuto devs having to design/test/maintain complicated P2P infrastructure.

**Comment by ghubstan** [comment]
>> 2. API > What help do you need to reduce the time of this task?

@ripcurlx has been using the api for personal stuff and that helps with testing.  @sqrrm has done all of the api reviews.  It will help if he continues.  They both have their own chores, so I'm not hoping they can take on much more than in the past.  Other devs said I was making steady progress before api work was put on hold.  I think I can return to that speed again pretty soon.

A fully functional api containing most/all features of UI is a long term task, and involves some refactoring (indirectly related to api, and not visible to end-users) to avoid duplicating code.  No ETA on delivering the "finished" api.  And adding some trading bots to the release is something I want to do, but we have to be cautious about it.  Up till now, the consensus was to let end users build and run their own bots at their own risk. It sounded kind of straight-forward and easy in the dev call, but much more discussion and prototyping is required. 

In the short term, I'm making sure the api will support creating swift payment accounts soon after @jmacxx's [PR](https://github.com/bisq-network/bisq/pull/5672) is merged.  I have it working in a local copy of his branch, and have asked him to review the swift account json-form (api analog to the ui's swift acct form).

I intend to start work on XMR <-> BTC trading w/ api tomorrow or Monday (06-Sept).

**Comment by pazza83** [comment]
Hi @leo816 

I would like to add 6-10 payment methods over the next couple of months.

I need help of a dev that is available to do the coding work required for adding them.

@jmacxx has done three payment methods, I am unsure as to how many more @jmacxx can take on?
@BtcContributor has done one but is no longer able to take on any work

It would be great if one or two more devs were available.

**Comment by leo816** [comment]
## Dev call 2021-09-10
### Topics

**1. Devin's Proposal**

- @devinbileck 's mobile improvement proposal https://github.com/bisq-network/proposals/issues/342 

hybrid app development approach, a single codebase that could support both, Swift app for IOS and java for android.
Possible developer candidate reached out.

concerns from Christoph: too many dependencies, is it easy to do an audit? potential security issue?

**2. API**

@ghubstan : Update:  about 3/4 done with work to support XMR/BTC trading pair in API. 

**3.How can we attract more developers?**
bisq is not running like a business so it's hard to hire people as if it were a business.

Stan: Maybe we could narrow down the search for web UI devs.   I can build a webapp over the core-api, and server Json. 
What could convince me to work on Bisq (beyond interesting content of the work) is confidence in the core devs' integrity (not scammers), and that I will be able to pay the bills.  Not necessarily a guaranteed paycheck, but a high level of confindence I will earn enough ducats to live on if my work is being accepted.

Also a consistent road-map... not too many detours on new projects that get suddenly dropped.  I know it can't be helped sometimes.  But detours should be taken thoughtfully.

**Comment by flybylow** [comment]
@Pazza has some great ideas, No one is assigned yet, feel free to put ideas out there. --> Where can I read up on the ideas? Would love to know more ...

**Comment by Pazza** [comment]
I appologize for posting this here, I can not find another way to reach out to @pazza83. For several months now, it appears as though you are getting tagged as @pazza which is notifying me of everything issue/comment/PR. Just an FYI to your users (as well as yourself who may not be receiving notifications).

**Comment by pazza83** [comment]
> I appologize for posting this here, I can not find another way to reach out to @pazza83. For several months now, it appears as though you are getting tagged as @Pazza which is notifying me of everything issue/comment/PR. Just an FYI to your users (as well as yourself who may not be receiving notifications).

Thanks for reaching out much appreciated, sorry that you are being notified. 

Hey everyone please can you tag me in to any discussion here as @pazza83 

My user name on Keybase is pazza but on here it is @pazza83

**Comment by pazza83** [comment]
> @Pazza has some great ideas, No one is assigned yet, feel free to put ideas out there. --> Where can I read up on the ideas? Would love to know more ...

Hi @flybylow  many thanks I will summarize the UX posts I have proposed yet to be implemented shortly and post here.

**Comment by leo816** [comment]
## Dev call 2021-09-20
### Topics



**1. Bisq on mobile**
Devin discussing the debate about what language to use for his proposal (Kotlin)
the difficulty finding the right devs for this.
m52go would like to see a firm timeline for the Mobile project to have interop with the Bisq network natively (e.g. the client/server access to offer book that chimp mentioned).

**2. Release 1.7.4** 
Emzy just pushed a pull request for the tor v3
tor v3 works. Have used it for a week.
tomorrow Chirstoph will likely merge. 
Friday likely prerelease.

**3. API and Daemon update**
@ghubstan API (gRPC Daemon+CLI) update:

I posted a warning in #general & #api about an API bug that could have shown a changing "buyer cost" value in Fiat/BTC trades (market-price-margin offers) in the CLI console's 'gettrade --trade-id' output.   So far, no API user has claimed they paid too much, but it's early.  
I posted a fix with an explanation of the problem in 
https://github.com/bisq-network/bisq/pull/5704

API (gRPC Daemon+CLI) 'editoffer' support has been merged into the master branch;  it might be included in the next Bisq release.

Consensus among core devs about what I should work on next would determine my short term priority.  Otherwise, I work on a new 'gettrades' API method. 

**4. Attracting new developers** (who may want to contribute on their free time but don't know about bisq).

- Chaincode labs
- Tomorrow's call with TBD


**5. Review of priority list**

<details>

<summary>Click to see new priority list</summary>

 these are Bisq's top priorities in the short-term:

 **1. Tor. v3:**

- we have @ripcurlx and @sqrrm working on this as the top priority.
- October 15th is the date v2 is disabled so we have to fix it before then
- It seems like its almost done

**2. API**

@ghubstan :
API (gRPC Daemon+CLI) support for trading xmr/btc is awaiting review.
I am re-writing the CLI console output formatting code to make it easier to use and maintain.  Almost done. 
tasks:
(a) Go back to work on simplifying Bisq packaging (but JReleaser is not quite ready).
(b) Add a new API method 'gettrades' that takes a currency-code param (at least), and returns trade history summary for a specific trading pair.
(c) Maybe, build a POC showing how a webapp could be built on top of the API to support thin web/mobile clients that GET and POST json.


**3. Messaging Problem**

- Some messages aren't going through, the reason is yet unclear.
- There seems to be a gap of information in the log. for 6 min. 
- Chimp mentioned the hybernate mode could be the cause for the messaging problem.
- PR to advise users when their PC has been in sleep mode (action item for @jmacxx)
- upon starting mediation, prompt users to expect a mediator response within 48 hours or contact them on keybase (action item for @jmacxx)

**4. Mobile app and UX**

- the Kotlin multiplatform is the best way going forward
- @Devinbilek setting some bounty's for basic tasks (recreating the app using pedro's desings)
- @m52go would like to see a firm timeline for the Mobile project to have interop with the Bisq network natively (e.g. the client/server access to offer book that chimp mentioned).
- 
**6.  XMR**

- Monero is our most traded altcoin, we should learn as much as possible and make sure we are doing what we can to attract as many users. these are the main complaints so far that users have reached out to us for:

1. If the auto-confirm is not activated, some sellers never respond and that can entail to having their funds locked for days. (are they all aware that they can activate this tool on settings?)
2. If the trade were to go to mediation, users have to write down their tx id and tx key to the mediator which is an annoying.
3. Given that trading on Bisq is tantamount to routinely checking if somebody hasn't lowered the price below your level it would be nice to have an option to one-click decresase or increase the offer price by 0.01% instead of manually entering the deviation when editing the offer.


**7.  Payment methods**

- 23 payment methods was the goal of which 9 have been added already.
- @jmacx has added 8 already
- @m52go had some concerns in evaluating the existing ones.
- Chimp tool to allow users to propose their own payment method at their own risk.
- also to improve the social aspect of bisq.
- do not risk the security reputation of bisq by adding unwanted payment methods.

**8.  Adding Tether**

- We should add tether, just choose the most used chain(Omni, ERC20 or TRC20)

**- Topics for next call:**

**Comment by aalmiray** [comment]
Hello everyone, this is Andres, author of JReleaser. In your own words, what's required to consider JReleaser as ready? Please feel free to post a discussion topic and/or issues at https://github.com/jreleaser/jreleaser

**Comment by leo816** [comment]
## Dev call 2021-09-28
### Topics


**1. X** 
@cbeams Q&A

**2. API update**
@ghubstan API  update:

I am working on a new API CLI getrades --category=<open|closed|failed> method.  Submission of a PR for that will be preceded by some preliminary refactoring work in a couple of desktop module trade views and models (another PR). I think it will be ready ~6-SEP, +/- a couple of days.

After that (bugs or other priorities permitting), I was planning to build a Bisq webapp POC that depends on Core API (no gRPC), and will serve json to thin web/mobile clients.  The POC would respond to browser HTTP REST requests by serving them raw json.  I will not be building a UI.

API (gRPC Daemon+CLI) 'editoffer' support has been merged into the master branch;  it might be included in the next Bisq release.

Consensus among core devs about what I should work on next would determine my short term priority.  Otherwise, I work on a new 'gettrades' API method. 


**3. Review of priority list**

<details>

<summary>Click to see new priority list</summary>

 these are Bisq's top priorities in the short-term:

 **1. Tor. v3:**

- Done.

**2. API**

I am working on a new API CLI getrades --category=<open|closed|failed> method.  Submission of a PR for that will be preceded by some preliminary refactoring work in a couple of desktop module trade views and models (another PR). I think it will be ready ~6-SEP, +/- a couple of days.

After that (bugs or other priorities permitting), I was planning to build a Bisq webapp POC that depends on Core API (no gRPC), and will serve json to thin web/mobile clients.  The POC would respond to browser HTTP REST requests by serving them raw json.  I will not be building a UI.


**3. Messaging Problem**

- Some messages aren't going through, the reason is yet unclear.
- There seems to be a gap of information in the log. for 6 min. 
- Chimp mentioned the hybernate mode could be the cause for the messaging problem.
- PR to advise users when their PC has been in sleep mode (action item for @jmacxx) **DONE**
- upon starting mediation, prompt users to expect a mediator response within 48 hours or contact them on keybase (action item for @jmacxx) **DONE**

**4. Mobile app and UX**

- @Devinbilek update:
- Assign bounties and whether it needs consensus.
- Once that is decided, it will be advertised on twitter to look for the right talent
- Also some tasks will require ux design so hoping Pedro can mock them up.


**6.  XMR**

- Monero is our most traded altcoin, we should learn as much as possible and make sure we are doing what we can to attract as many users. these are the main complaints so far that users have reached out to us for:

1. If the auto-confirm is not activated, some sellers never respond and that can entail to having their funds locked for days. (are they all aware that they can activate this tool on settings?)
2. If the trade were to go to mediation, users have to write down their tx id and tx key to the mediator which is an annoying.
3. Given that trading on Bisq is tantamount to routinely checking if somebody hasn't lowered the price below your level it would be nice to have an option to one-click decresase or increase the offer price by 0.01% instead of manually entering the deviation when editing the offer.


**7.  Payment methods**
- @jmacx continuing to make progress,
- 23 payment methods was the goal of which 9 have been added already.


**8.  Adding Tether**

- @pazza83 already created the proposal.  https://github.com/bisq-network/proposals/issues/170#issuecomment-927207827
- We should add tether, just choose the most used chain(Omni, ERC20 or TRC20)

**- Topics for next call:**

**Comment by leo816** [comment]
**Priority list**

 **1. BSQ<->BTC swap**

- @ripcurlx  annd @sqrrm are on this.
- the chimp also joined Sqrrm to review the new protocol, @ripcurlx has  switched to continue the work of Jakub-CZ with the buy-BSQ changes.

**2. API**

I am working on a new API CLI getrades --category=<open|closed|failed> method.  Submission of a PR for that will be preceded by some preliminary refactoring work in a couple of desktop module trade views and models (another PR). I think it will be ready ~6-SEP, +/- a couple of days.

After that (bugs or other priorities permitting), I was planning to build a Bisq webapp POC that depends on Core API (no gRPC), and will serve json to thin web/mobile clients.  The POC would respond to browser HTTP REST requests by serving them raw json.  I will not be building a UI.


**3. Messaging Problem**

- Some messages aren't going through, the reason is yet unclear.
- There seems to be a gap of information in the log. for 6 min. 
- Chimp mentioned the hybernate mode could be the cause for the messaging problem.
- PR to advise users when their PC has been in sleep mode (action item for @jmacxx) **DONE**
- upon starting mediation, prompt users to expect a mediator response within 48 hours or contact them on keybase (action item for @jmacxx) **DONE**

**4. Mobile app and UX**

- @Devinbilek update:
- Assign bounties and whether it needs consensus.
- Once that is decided, it will be advertised on twitter to look for the right talent
- Also some tasks will require ux design so hoping Pedro can mock them up.


**6.  XMR**

- Monero is our most traded altcoin, we should learn as much as possible and make sure we are doing what we can to attract as many users. these are the main complaints so far that users have reached out to us for:

1. If the auto-confirm is not activated, some sellers never respond and that can entail to having their funds locked for days. (are they all aware that they can activate this tool on settings?)
2. If the trade were to go to mediation, users have to write down their tx id and tx key to the mediator which is an annoying.
3. Given that trading on Bisq is tantamount to routinely checking if somebody hasn't lowered the price below your level it would be nice to have an option to one-click decresase or increase the offer price by 0.01% instead of manually entering the deviation when editing the offer.


**7.  Payment methods**
- @jmacx continuing to make progress,
- 23 payment methods was the goal of which 9 have been added already.


**8.  Adding Tether**

- @pazza83 already created the proposal.  https://github.com/bisq-network/proposals/issues/170#issuecomment-927207827
- We should add tether, just choose the most used chain(Omni, ERC20 or TRC20)

**- Topics for next call:**

**Comment by leo816** [comment]
## Dev call 2021-10-15
### Topics

**1. BSQ<->BTC swap** 
Almost done, next release will include it.
@ripcurlx :
making the whole process easier (specially for non users)

**2. Payment Methods** 
@jmacxx added 14 new payment methods.

**3. API update**
@ghubstan API  update:



**4. XMR/BTC swap** 
Currently reviewing the process to implement it.

**5. Review of priority list**

<details>

<summary>Click to see new priority list</summary>

 these are Bisq's top priorities in the short-term:

**1. BSQ<->BTC swap** 
Almost done, next release will include it.
@ripcurlx :
making the whole process easier (specially for non users)

**2. API**
@ghubstan 
 After atomic-tx is merged, I'll follow up on adjustments in the API (to support atomic-tx), 

Any work on a Bisq/WebApp POC will have pushed back to sometime in November or December.

 XMR:  API work to support XMR / BTC trd pair was done, but also being held back until after atomic-tx work is merged. Then I will re-work that to. 


**3. Messaging Problem**

Possible solution is included in the next release.


**4. Mobile app and UX**

- @Devinbilek update:
- Defined all of the tasks that need to be addressed.
- Working through the UX designs that pedro made.
- some screens and workflow are missing
- reached out to the Bitcoin design community. 1 person is interested.
- previous app developer is interested in helping with this one. (joaquin)

**6.  XMR**

- Monero is our most traded altcoin, we should learn as much as possible and make sure we are doing what we can to attract as many users. these are the main complaints so far that users have reached out to us for:

1. If the auto-confirm is not activated, some sellers never respond and that can entail to having their funds locked for days. (are they all aware that they can activate this tool on settings?)
2. If the trade were to go to mediation, users have to write down their tx id and tx key to the mediator which is an annoying.
3. Given that trading on Bisq is tantamount to routinely checking if somebody hasn't lowered the price below your level it would be nice to have an option to one-click decresase or increase the offer price by 0.01% instead of manually entering the deviation when editing the offer.


**7.  Payment methods**
- @jmacx continuing to make progress,
- 14 payment methods will be added.


**8.  Adding Tether**

- @jmacxx already implemented it. will be included in the next release.

**9. XMR/BTC swap** 
Currently reviewing the process to implement it.

**- Topics for next call:**

**Comment by ghost** [comment]
## Dev call 2021-10-29

### Topics & Priorities

- **Bisq mobile**: new potential contributor to be put into contact with Devin.

- **BSQ<->BTC atomic swap**
PRs and refactorings done, being reviewed by @stejbac, @ghubstan, @ripcurlx et al.

- **DAO performance improvement / refactor**
PRs ready for review & testing.  Action all devs.

- **v1.8.0** - We should allocate more time for testing this cycle due to the major changes.  Several more days would be good, so that full regression testing can be performed.  Action @ripcurlx @BtcContributor

- @ripcurlx now has immunity to all viruses (:crossed_fingers: )

- **API**: @ghubstan to continue.

- **Running Bisq on Umbrel** - @ghubstan to consider investigating.

- **Discussion of** https://github.com/bisq-network/proposals/issues/344
Action item for all: give feedback.
See if a small scale project can be defined, with suitable budget.  Project better focus than 'role'.
@pazza83 - One possible project idea would be to solicit input from XMR users.

- **Payment methods** - not mentioned in the meeting, but there are a [few more left to add](https://github.com/bisq-network/bisq/discussions/5462). Action @jmacxx

- @jimmyneutront expressed the desire to hear more about **X Protocol** progress.

**Comment by leo816** [comment]
## Dev call 2021-11-08
### Topics

**1. BSQ<->BTC swap** 
PRs and refactorings done, being reviewed by @stejbac, @ghubstan, @ripcurlx et al.

**2. Payment Methods** 
@jmacxx : 
working on the last 4 payment methods for @pazza; started on Grin re-listing tasks; and the usual bug fixing / support.

**3. API update**
@ghubstan API  update:
re-submitted some API related work in the last few days: 
(1) some minor adjustments in the API to the new bsq-swap support,
(2) improvements to the API CLI's output formatting code, to make it easier to use as new API methods are added.  
(3) BSQ swap testing, looking at umbrel
 

**4. XMR/BTC swap** 
@ghubstan XMR/BTC trading pair support in in the API -- adjusted for the new bsq-swap feature.

**5. Research & Connectivity** 
Chimp - waiting for PRs to be merged; did some research into Misq & Kotlin / async.  Looking into tor connectivity issues.

**6. Review of priority list**

<details>

<summary>Click to see new priority list</summary>

 these are Bisq's top priorities in the short-term:

**1. BSQ<->BTC swap** 

@ripcurlx :
Reviewing @chimp1984's PRs  and working on the Buy BSQ branch that was merged today. 

 all - getting the PRs reviewed and merged so there is ample time for system testing of release 1.8.0.

**2. API**
@ghubstan 
BSQ swap testing, 
Next contributions will include re-working XMR/BTC trading pair support in in the API -- adjusted for the new bsq-swap feature.
Work on a new API gettrades method to display lists of OPEN|CLOSED|FAILED trades in the CLI console,


**3. Messaging Problem**

Waiting for more bisq logs to better understand this issue


**4. Mobile app and UX**

- @Devinbilek update:
waiting to hear from potential dev contributor via @ripcurlx

**5.  XMR**

- XMR/BTC swap 
@ghubstan XMR/BTC trading pair support in in the API -- adjusted for the new bsq-swap feature.

 Monero is our most traded altcoin, we should learn as much as possible and make sure we are doing what we can to attract as many users. these are the main complaints so far that users have reached out to us for:

1. If the auto-confirm is not activated, some sellers never respond and that can entail to having their funds locked for days. (are they all aware that they can activate this tool on settings?)
2. If the trade were to go to mediation, users have to write down their tx id and tx key to the mediator which is an annoying.
3. Given that trading on Bisq is tantamount to routinely checking if somebody hasn't lowered the price below your level it would be nice to have an option to one-click decresase or increase the offer price by 0.01% instead of manually entering the deviation when editing the offer.


**6.  Payment methods**
@jmacxx : 
working on the last 4 payment methods for @pazza; started on Grin re-listing tasks; and the usual bug fixing / support.


**7. XMR/BTC swap** 
Currently reviewing the process to implement it.

**- Topics for next call:**

**Comment by leo816** [comment]
## Dev call 2021-12-07
### Topics

**1. Provide ASIC resistant PoW scheme for BSQ swaps** 
Already merged by Steven. Now he will focus on bug fixing
https://github.com/bisq-network/bisq/pull/5858

**2. sqrrm started reviewing the API PRs**

**3. Messaging problem**
Is the core to a large percentage of all failed trades and yet no one is looking into it besides James

**4. Review of priority list**

<details>

<summary>Click to see new priority list</summary>

 these are Bisq's top priorities in the short-term:

**1. Release testing for 1.8.0** 
@ripcurlx :
running the release branch for a couple of days
then will go back to bisq monitor fixing.

**2. misq/bisq2.0** 
Chimp is back to working on bisq 2.0 

**3. API**
@ghubstan :
Back to work for the next two weeks on getting API working for BSQ swaps, and re-working the API XMR/BTC trading pair support on top of recently merged BSQ swap support.


**4. Messaging Problem**
James will be sending sqrrm and Steven logs to study what happens here. 

@ripcurlx found an error log that may be related to the messaging problem, specifically regarding it having an incorrect public key shown. more information will follow.


**5. Mobile app and UX**
@Devinbilek update:
Still trying to build out an initial project structure and waiting on designer to build out views/workflows. I have not made progress lately as I have been very busy.

**6.  XMR**
It takes a lot of CPU and users are complaining a lot that bisq is slow.
chimp is taking care of this, creating the option to have a light mode or full mode for heavy users.
receiving all the hashes but only verifying the latest block.

- XMR/BTC swap 
@ghubstan Back to work for the next two weeks re-working the API XMR/BTC trading pair support on top of recently merged BSQ swap support.


**8.  Payment methods**
@pazza83  will be in charge of the reporting regarding the performance of the recently added payment methods.


I started reviewing the API PRs, they should be fairly easy and it needs to be done 
**- Topics for next call:**

**Comment by leo816** [comment]
## Dev call 2022-01-10
### Topics


**1. Misq brainstorm and updates** 
Update of current plans and review of in progress work from Chimp.
It would also be great to have someone developing the mobile app simultaneously. No updates from Devin yet.

**2. Recommendation for power users**
for big bisq users, using a local bitcoin node will really help and avoid different bugs.
creating a lot of offers loads too much your client.
20/30 offerrs at once is the limit as to what could be generating problems.

**3. We need a full time UI developer**
working for the next 4-5 months
It would also be great to have someone develop

**4. Growth**
BSQ/BTC Swap Have been adopted nicely but no marketing and virtually no one knows about it.
Team lead for growth is key.

**5. Review of priority list**

<details>

<summary>Click to see new priority list</summary>

 these are Bisq's top priorities in the short-term:

**1. bisq monitor fixing** 
@ripcurlx currently top priority.
API examples documentation. to show that the API is fully functional, have 3 examples that companies can embrace.

**2. misq/bisq2.0** 
@chimp's main focus.
2 options: simple remote with full node, (connect with the mobile)/ and the more advance to have an almost full node to which you connect. 1 tor connection to the network, easier to run it on android.


**3. API**
@ghubstan : 
API:  create/take offers, finish v1 and bsq-swap trades, create fiat & xmr payment accts. 
ghubstan says:But no meidation / arbitration support yet. 


**4. Mobile app and UX**
@Devinbilek : No update on the mobile app. Had no time to work on it recently.


**5.  XMR**
It takes a lot of CPU and users are complaining a lot that bisq is slow.
chimp is taking care of this with bisq 2.0, creating the option to have a light mode or full mode for heavy users.
receiving all the hashes but only verifying the latest block.


**- Topics for next call:**

**Comment by leo816** [comment]
## Dev call 2022-02-11
### Topics


**1. Bisq2.0/Misq brainstorm and updates** 
Update of current plans and review of progress work from Chimp.
Connecting from i2p y Tor simultaneously.
Liquid for multisig and swaps will work. 
social aspect -  moderators will have to burn btc and will be compensated.


**2. We need a full time UI developer**
working for the next 4-5 months
It would also be great to have someone develop the mobile app.

**3. Growth/marketing**
BSQ/BTC Swap Have been adopted nicely but no marketing and virtually no one knows about it.
Team lead for growth is key and people engaging more on platforms like reddit is key.

**4. Liquid implementation**
Alva already implemented it this week.

**5. Updates from last call**
"API examples documentation. to show that the API is fully functional, have 3 examples that companies can embrace." 
first examples from ripcurl: https://github.com/ripcurlx/bisq-api-examples
@ghubstan :
I'm still working on api reference doc generations, then commenting api protobuf files (api reference doc content), then snippets for api calls, then the "3 examples" (complete bot trading scripts) 

**6. Running the numbers**
Pazza's monthly P/L review: https://github.com/bisq-network/roles/issues/111#issuecomment-1029551837


**6. Review of priority list**

<details>

<summary>Click to see new priority list</summary>

 these are Bisq's top priorities in the short-term:


**1. API** 
n1 priority. It will boost volume.
@ghubstan says:
What I am doing may increase Bisq1 API usage, bringing additional funds to support Bisq2 work.

I am working on a stand-alone bisq-apidoc project to parse pb.proto and grpc.proto files into Markdown for a Slate
based gRPC API Reference document site.
Here is a sample: https://ghubstan.github.io/slate

After bisq-apidoc is ready for review (probably next week), I will start commenting the Bisq repo's pb.proto
and grpc.proto files. Those .proto file comments become the descriptions of gRPC services, methods, messages, fields
and enums displayed in the gRPC API Reference doc.

The next phase involves several tasks, not necessarily done in the following order or serially:

1.  (a) Work with reviewers of the [WIP] API Document to find protobuf definitions that need to change to make the
    gRPC API more usable and intuitive -- not worrying about backward compatibility.

    (b) Make appropriate changes to the API, update API Reference doc, repeat...

2.  Start creating code snippets showing how each API method is called from the CLI, Java, and Python.
    The code snippets will appear to the right of each gRPC API method described in the API Reference doc.
In the longer term:

3.  Work with others (ripcurlx, sqrrm, pazza, ?) to build simple Python trading bot scripts.

4.  Work with @m52go and others to figure out how to integrate a more polished Bisq1 gRPC API into
    interested businesses.


**2. bisq monitor fixing** 
@ripcurlx 's top priority from last call, still waiting for an update before editing this.

**3. misq/bisq2.0** 
@chimp's main focus.

next goals for bisq2:

- reputation based trading protocol
- Monero btc swaps
- lighting implementation (3 party based trading protocol, similar to the function of the multisig)


@jimmyt:  some simple python snippets showing how to call the api / rpc methods in 
https://ghubstan.github.io/slate
 would help a lot.  Another guy named @langfield may also help out. 


**4. Mobile app and UX**
@Devinbilek : No update on the mobile app. Had no time to work on it recently.
It makes more sense to tackle directly the mobile app for bisq 2.0


**5.  XMR**
Monero/btc swaps is one of the main goals, right now no one is on top of this.


**- Topics for next call:**

**Comment by leo816** [comment]
## Dev call 2022-02-02
### Topics


**1. Comments on Tasks/priorities** 
ripcurlx priorities: review/merge PRs, esp. API to "market" it with users; then UI + contact with UI devs (no leads for now); API is good enough for trading bot, not for mediation/arbitration; might be issue where API daemon needs restarting with heavy usage.

**2. Next release**
ripcurlx: release 1.8.3 close as it just needs some work on the DAO state fix/restart situation.

**3. Growth/marketing**
BSQ/BTC Swap Have been adopted nicely but no marketing and virtually no one knows about it.
Team lead for growth is key and people engaging more on platforms like reddit is key.

**4. Concerns**
 i. jmacxx is concerned that the messaging problem will be inherited by bisq2.
reports issue with messaging over network failing in ~1% of cases, maybe Tor problem, no clues as of now how to solve; alternative trading implementation, robosats over LN.
ii. pazza: citing m52go, with bisq2 it's important to keep contributors' privacy, but maybe also keep private trades information (and profit/loss info as well?)
(Create a guideline on how to "keep" a private profile as contributor.)
w000000t: agreeing to write a wiki article about privacy for contributors


**5. Review of priority list**

<details>

<summary>Click to see new priority list</summary>

 these are Bisq's top priorities in the short-term:


**1. API** 
ripcurlx will also have this as n1 priority.
Except the mediation arbitration everything is working with the current api.

@ghubstan update for dev call 2-Mar-2022:

I've been working on an API Reference doc generator:  https://github.com/ghubstan/bisq-grpc-api-doc
From recently merged .proto file comments, it produces:  https://ghubstan.github.io/slate .
There are CLI / Java / Python examples for each RPC method, e.g.,
	https://ghubstan.github.io/slate/?shell#rpc-method-createoffer
	https://ghubstan.github.io/slate/?java#rpc-method-createoffer
	https://ghubstan.github.io/slate/?python#rpc-method-createoffer
If/when reviewers/maintainers agree it is in good enough shape, we just need to fork https://github.com/slatedocs/slate into a new https://github.com/bisq-network repo, and add changes via slate fork PRs.  Deploying changes to a single .md file is simple.

I have started work on my first Python API bot example:  migrating ripcurlx's 
https://github.com/ripcurlx/bisq-api-examples/blob/main/cli/best_price_offers.py to 
https://github.com/ghubstan/bisq-grpc-api-doc/blob/main/src/main/python/examples/bots/best_priced_offer_bot.py

After this 'best' -- or rather, most 'competitive' -- price bot is deemed OK, I'll work on a market-maker bot.
Other bot example ideas will come from others: flix, pazza, m52go, https://kollider.medium.com/build-a-crypto-market-making-bot-in-python-d71eeae2dcd7, etc...

There is a guy from BTCPay named Wouter Samaey who wanted to talk about API integration issues, and now that there's an API reference and progress on Python bots, I'm more comfortable chatting with people like him about integration issues.  I'll try to convince him that investigating Bisq API integration is worth his time.  

Last thing I want to say here is that the API users are going to bump into problems with the bitcoinj wallet.  There is a hard limit on the number of v1 trade protocol trades that can be made during a single API daemon session: I'm pretty sure it is 33 trades.  (Sqrrm and I struggled with this during his work on the BSQ swap changes.)  I don't know how to work around that without shutting down and restarting the server, and I suspect it will be [EDIT "one of"] the most annoying problems for any businesses wanting to integrate the Bisq API.



**2. misq/bisq2.0** 
Chimp Right now on holiday.
Christoph will take care of the json library which right now doesn't support java 5 and above.
2 new part time developers are interested

- alva: working on Bisq2 with chimp; working on liquid with stablecoins.


next goals for bisq2:
- UI for liquid wallet
- lightweight wallet
- PR to use JSON to query bitcoin in Bisq2
https://github.com/bisq-network/projects/issues/64
 

**3. Mobile app and UX**
@Devinbilek : No update on the mobile app. Had no time to work on it recently.
It makes more sense to tackle directly the mobile app for bisq 2.0

**4.  XMR**
Monero/btc swaps is one of the main goals, right now no one is on top of this.


**5. bisq monitor fixing** 
@ripcurlx not working on this right now because there are higher priorities.

**Comment by leo816** [comment]
## Dev call 2022-04-20
### Topics

**1. Comments on Tasks/priorities** 
Bisq2.0 update with chimp. will take about 2 months to have a first prototype ready. alcom, alva will be joining as devs.
the chat based area should be very feasible to adapt to mobile.
promoting bisq is crucial.
chimp: API should still be n1 on the priority list. 

security model in terms of how the offers are secured, reputation concept based to bring everything 0 and 1. this number will be calculated directly in the background, when you create an identity it has a timestamps and that is a small part of reputation.

**2. Growth/marketing**
We have been in contact with many people to try and help bisq grow. We are still looking for someone to speak in the name of bisq.


**3. Review of priority list**

<details>

<summary>Click to see new priority list</summary>

 these are Bisq's top priorities in the short-term:


**1. API** 

navigation changes have been made, 
we can start with release testing, will be shipped and documentation. more easily accesible.
still waiting for stan, he is not available right now, waiting for the bitcoinpay server to get more insight.
expecting 

it won't be transferable to bisq2.0 

ripcurlx will also have this as n1 priority.
Except the mediation arbitration everything is working with the current api.


next goals for bisq2:
- UI for liquid wallet
- lightweight wallet
- PR to use JSON to query bitcoin in Bisq2
https://github.com/bisq-network/projects/issues/64
 

**2. Mobile app and UX**
No update on mobile, this will be a high priority as soon as bisq2 is ready

**3.  XMR**
Monero/btc swaps is one of the main goals, right now no one is on top of this.


**4. bisq monitor fixing** 
@ripcurlx not working on this right now because there are higher priorities.

**Comment by leo816** [comment]
## Dev call 2022-06-24
### Topics

**1. Comments on Tasks/priorities** 
- Tor issue seems partly solved but future problems will likely arise as there is no way to solve it from our side. Useful to implement at least an additional tor monitor to have multiple views of the network status, but long term solution is having multiple supported networks(We have to make sure we always have the latest release)

- Emzy: Price nodes Issue, some exchanges had some weird prices. not sure how to manage that. If its off more than 2% then we should do something to prevent that. We had an experience where the monero price was diverging but onnly by a small percentage, but brazillian dollar was way off. Manfred suggests that we can implement some code to detect outliers.
- tor issues and seednodes not working and then working again by themselves; implement seednodes switching when one stops responding
- Haveno stopping UI dev 

**2. Bisq 2.0**
Chimp showing developments on bisq2. and giving a tutorial on how the new version will work. Full flow from login/user creation up to trading and inviting mediator, inline knowledge base; possibility of electrum wallet integration
Bisq 2.0 looks amazing, looking forward to the release. 

**3. Growth/marketing**
We have been in contact with many people to try and help bisq grow. 

**4. Review of priority list**

<details>

<summary>Click to see new priority list</summary>

 these are Bisq's top priorities in the short-term:


**1. API** 
Stan has been working on some bots, also has in mind the subprojects requested by chimp, he will take them on as soon as he is done with what he is doing now.


Now working on some java API bots these past few weeks, and just checked them into the https://github.com/bisq-network/bisq-api-reference project.  There is still some polishing left to do in the code, but they're ready to run on mainnet.  Next thing for me to do is explain them on the API reference site.

New API java bots are at the bottom:
https://github.com/bisq-network/bisq-api-reference/tree/main/java-examples/src/main/java/bisq/bots

Explanation for the TakeBestPricedOfferToSellBtc bot is here:
https://github.com/bisq-network/bisq-api-reference/blob/329e39c3c407999cf121871a06fad86a32c1315f/java-examples/src/main/java/bisq/bots/TakeBestPricedOfferToSellBtc.java#L38


**2. Mobile app and UX**
This will be a high priority as soon as bisq 2.0 is ready.

**3.  XMR**
Monero/btc swaps is one of the main goals, right now no one is on top of this.

**4. bisq monitor fixing** 
Emzy: monitor is not totally broken, data is coming directly.
@ripcurlx not working on this right now because there are higher priorities.

**Comment by pazza83** [comment]
Closing as stale
